libs/dmlf/include/dmlf/colearn/muddle_learner_networker_impl.hpp

```diff
@@ -109,23 +113,20 @@ class MuddleLearnerNetworkerImpl : public AbstractMessageController
     return update_store_->GetUpdateCount();
   }
 
-  ConstUpdatePtr GetUpdate(Algorithm const & algo = "algo0",
-                           UpdateType const &type = "gradients") override
+  ConstUpdatePtr GetUpdate(AlgorithmClass const &algo = "algo0",
+                           UpdateType const &    type = "gradients") override
   {
     return update_store_->GetUpdate(algo, type);
   }
 
-  void PushUpdateBytes(const std::string &type_name, Bytes const &update);
-  void PushUpdateBytes(const std::string &type_name, Bytes const &update, Peers peers);
+  void PushUpdateBytes(UpdateType const &type_name, Bytes const &update);
+  void PushUpdateBytes(UpdateType const &type_name, Bytes const &update, const Peers &peers,
+                       double broadcast_proportion = -1.0);
```

Comment:
-1.0?

Reply:
A number that isn't sensible, basically... I'd have used NaN, but actually making one is a pain.